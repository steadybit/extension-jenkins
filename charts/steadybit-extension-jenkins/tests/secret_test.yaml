templates:
  - secret.yaml
tests:
  - it: no secret with existing secret
    set:
      jenkins:
        apiToken: null
        existingSecret: existing-secret
    asserts:
      - hasDocuments:
          count: 0
  - it: manifest should match snapshot with api key
    set:
      jenkins:
        apiToken: 111-222-333
        existingSecret: null
    asserts:
      - matchSnapshot: { }
  - it: no secret without api key
    set:
      jenkins:
        apiToken: null
        existingSecret: null
    asserts:
      - hasDocuments:
          count: 0
  - it: should render nothing with default values
    asserts:
      - hasDocuments:
          count: 0
  - it: secret should be created with apiToken even when insecureSkipVerify is enabled
    set:
      jenkins:
        apiToken: 111-222-333
        existingSecret: null
        insecureSkipVerify: true
    asserts:
      - matchSnapshot: { }
      - isKind:
          of: Secret
      - equal:
          path: data["api-token"]
          value: MTExLTIyMi0zMzM= # Base64 encoded 111-222-333
