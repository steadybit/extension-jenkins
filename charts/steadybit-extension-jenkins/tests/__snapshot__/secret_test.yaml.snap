manifest should match snapshot with api key:
  1: |
    apiVersion: v1
    data:
      api-token: MTExLTIyMi0zMzM=
    kind: Secret
    metadata:
      labels: null
      name: steadybit-extension-jenkins
      namespace: NAMESPACE
    type: Opaque
secret should be created with apiToken even when insecureSkipVerify is enabled:
  1: |
    apiVersion: v1
    data:
      api-token: MTExLTIyMi0zMzM=
    kind: Secret
    metadata:
      labels: null
      name: steadybit-extension-jenkins
      namespace: NAMESPACE
    type: Opaque
